modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -200,11 +201,11 @@ public OperationException onError(OperationException error) {
     @GET
     @Path("{executionId}/status")
     public Object status(@PathParam("executionId") String executionId) throws IOException, MessagingException {
+        String error = getError(executionId);
```

Comment:
So now each normal call to /status costs 2 round trips, one to the transient store to check for an error and one to the kv to get the status.

Reply:
The whole problem comes from not checking errors on incomplete async operations. We need to make sure there is no error completed or not.
Not a dream state but 2 calls vs 1 call is still constant complexity-wise 🤷

Reply:
when `setError` is called is also calls `setCompleted`, in which case do we have errors with an execution not marked as completed ?

Reply:
when failing while compiling the operation chain, isCompleted is never called. Since it conditions the call to getErrors() we never see the error. So we made the call to getErrors independant and preceding isCompleted(): https://github.com/nuxeo/nuxeo/pull/3986/commits/028ad6c66f4120654dfdaec4b0391cde20220e1f#diff-632d12880ac130d2edc7fa897f2779eaR205-L207